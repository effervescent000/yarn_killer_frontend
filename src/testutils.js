import { MemoryRouter, BrowserRouter } from "react-router-dom";

import { UserContext } from "./user-context";

const wrapWithMemoryRouter = (ui) => {
    return <MemoryRouter>{ui}</MemoryRouter>;
};

const wrapWithBrowserRouter = (ui, { route = "/" } = {}) => {
    window.history.pushState({}, "test", route);
    return <BrowserRouter>{ui}</BrowserRouter>;
};

const wrapWithUserContext = (ui, { user = {}, loggedIn = false } = {}) => {
    return (
        <UserContext.Provider
            value={{ loggedIn, toggleLogIn: jest.fn(), user, setUser: jest.fn() }}
        >
            {ui}
        </UserContext.Provider>
    );
};

const yarnArray = [
    {
        // this one should have 1+ image and 1+ link
        brand: "Red Heart",
        color_style: "",
        colorways: [],
        discontinued: false,
        fibers: [
            {
                amount: 100,
                id: 1,
                type: "Acrylic",
                yarn_id: 1,
            },
        ],
        gauge: 17,
        id: 1,
        images: [
            {
                colorway_id: 0,
                id: 1,
                label: null,
                url: "https://images4-g.ravelrycache.com/uploads/redheartyarns/797884997/upload",
                yarn_id: 1,
            },
        ],
        links: [
            {
                id: 10,
                url: "www.michaels.com/aegjnakg",
                store_id: 5,
                yarn_id: 1,
                current_price: null,
                price_updated: null,
                store: "Michael's",
            },
        ],
        name: "Super Saver Solids",
        texture: "Cabled",
        weight_grams: 198,
        weight_name: "Aran",
        yardage: 364,
    },
    {
        // this one should have no images
        brand: "Caron",
        color_style: "",
        colorways: [],
        discontinued: false,
        fibers: [
            {
                amount: 100,
                id: 2,
                type: "Acrylic",
                yarn_id: 2,
            },
        ],
        gauge: 18,
        id: 2,
        images: [],
        links: [],
        name: "Simply Soft Solids",
        texture: "Plied (3+)",
        weight_grams: 170,
        weight_name: "Aran",
        yardage: 315,
    },
];

export { wrapWithMemoryRouter, yarnArray, wrapWithBrowserRouter, wrapWithUserContext };
