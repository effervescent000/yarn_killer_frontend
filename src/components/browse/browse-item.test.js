import renderer from "react-test-renderer";

import BrowseItem from "./browse-item";
import { routerWrapper } from "../../testutils";

const yarnArray = [
    {
        brand: "Red Heart",
        color_style: "",
        colorways: [],
        discontinued: false,
        fibers: [
            {
                amount: 100,
                id: 1,
                type: "Acrylic",
                yarn_id: 1,
            },
        ],
        gauge: 17,
        id: 1,
        images: [
            {
                colorway_id: 0,
                id: 1,
                label: null,
                url: "https://images4-g.ravelrycache.com/uploads/redheartyarns/797884997/upload",
                yarn_id: 1,
            },
        ],
        links: [],
        name: "Super Saver Solids",
        texture: "Cabled",
        weight_grams: 198,
        weight_name: "Aran",
        yardage: 364,
    },
    {
        brand: "Caron",
        color_style: "",
        colorways: [],
        discontinued: false,
        fibers: [
            {
                amount: 100,
                id: 2,
                type: "Acrylic",
                yarn_id: 2,
            },
        ],
        gauge: 18,
        id: 2,
        images: [],
        links: [],
        name: "Simply Soft Solids",
        texture: "Plied (3+)",
        weight_grams: 170,
        weight_name: "Aran",
        yardage: 315,
    },
];

describe("BrowseItem tests", () => {
    test("Renders with image", () => {
        const yarn = yarnArray[0];
        const tree = renderer.create(routerWrapper(<BrowseItem yarn={yarn} />)).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Renders without image", () => {
        const yarn = yarnArray[1];
        const tree = renderer.create(routerWrapper(<BrowseItem yarn={yarn} />)).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
