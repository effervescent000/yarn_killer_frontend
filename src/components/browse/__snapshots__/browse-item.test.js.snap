// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseItem tests Renders with image 1`] = `
<div
  className="yarn-item"
>
  <a
    href="/yarn/1"
    onClick={[Function]}
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="Red Heart Super Saver Solids"
        src="https://images4-g.ravelrycache.com/uploads/redheartyarns/797884997/upload"
      />
    </div>
    <div
      className="text-wrapper"
    >
      <div
        className="yarn-brand"
      >
        Red Heart
      </div>
      <div
        className="yarn-name"
      >
        Super Saver Solids
      </div>
    </div>
  </a>
</div>
`;

exports[`BrowseItem tests Renders without image 1`] = `
<div
  className="yarn-item"
>
  <a
    href="/yarn/2"
    onClick={[Function]}
  >
    <div
      className="image-wrapper"
    />
    <div
      className="text-wrapper"
    >
      <div
        className="yarn-brand"
      >
        Caron
      </div>
      <div
        className="yarn-name"
      >
        Simply Soft Solids
      </div>
    </div>
  </a>
</div>
`;
